lib/composite.ex

```diff
@@ -387,6 +406,21 @@ defmodule Composite do
     {query, MapSet.union(loaded_deps, deps_to_load)}
   end
 
+  defp maybe_raise_on_unknown_params(%__MODULE__{strict: true} = composite) do
+    diff =
+      MapSet.difference(
+        MapSet.new(Map.keys(composite.params)),
```

Comment:
`composite.params` is not restricted to map only. It can be anything that implements `Access` behaviour.
`Access` module doesn't have anything that can list keys for the term. I suggest applying this logic to maps and keywords only, and ignore structs. Like,
```elixir
if is_map(composite.params) and not is_struct(composite.params) or Keyword.keyword?(composite.params) do
  # ...
else
  :noop
end
```
With this approach, `Enum` module can be used to navigate through the params

Reply:
Done!